server/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -376,11 +379,15 @@ private static ResolveResult resolve(String pathElement, String fullPath, Object
         if (context == null) {
             return ResolveResult.error("cannot resolve [" + pathElement + "] from null as part of path [" + fullPath + "]");
         }
-        if (context instanceof Map<?, ?> map) {
-            if (map.containsKey(pathElement)) {
-                return ResolveResult.success(map.get(pathElement));
+        if (context instanceof Map<?, ?>) {
+            @SuppressWarnings("unchecked")
+            Map<String, Object> map = (Map<String, Object>) context;
+            Object object = map.getOrDefault(pathElement, NOT_FOUND); // getOrDefault is faster than containsKey + get
```

Comment:
I assume we can't do `map.get(pathElement)` and treat a `null` value as "not found" because a field might actually have the `null` value?

Reply:
Indeed, we don't control these maps in a meaningful way so null values are allowed.